xamarin/Xamarin.Forms
# Fix RTL on Labels with FormattedText and/or LineHeight

### Description of Change ###
Fix RTL so that it works on labels with FormattedText and/or LineHeight
Fix RTL so that it works on labels with other previously incompatible properties as well (i.e. FontAttributes, FontSize, LineHeight, TextColor, TextTransform, TextType, VerticalTextAlignment)

### Issues Resolved ### 
- fixes #3311 
- fixes #12473
- fixes #8157

### Platforms Affected ### 
- iOS

### Behavioral/Visual Changes ###
Labels with RTL (and with FormattedText and/or LineHeight) are properly right-aligned now

### Before/After Screenshots ### 

|   | **BEFORE** |**AFTER**   | 
|---|---|---|
|  Label w/ RTL and FormattedText | ![Simulator Screen Shot - iPod touch (7th generation) - 2021-01-15 at 14 57 36](https://user-images.githubusercontent.com/21988533/104774209-46136180-5744-11eb-9fb3-4606cb1c0732.png) | ![Simulator Screen Shot - iPod touch (7th generation) - 2021-01-15 at 14 56 26](https://user-images.githubusercontent.com/21988533/104774240-4f9cc980-5744-11eb-87a2-a6f96a43eef5.png) |
| Label w/ RTL and LineHeight  |  ![Simulator Screen Shot - iPod touch (7th generation) - 2020-12-18 at 16 38 37](https://user-images.githubusercontent.com/21988533/102664976-eed49e00-4151-11eb-9f6c-4c6fae6abad6.png) | ![Simulator Screen Shot - iPod touch (7th generation) - 2020-12-18 at 16 43 49](https://user-images.githubusercontent.com/21988533/102664990-f5fbac00-4151-11eb-99d3-e9d1033b9f42.png)  |

### Testing Procedure ###
- Run `Issue3311`
- Run `Issue12743`
- Make sure `RTLLabelWithFormattedTextWorks` and `RTLLabelWithLineHeightWorks` tests are passing

### PR Checklist ###
